core-services/src/main/java/org/zalando/nakadi/partitioning/HashPartitionStrategy.java

```diff
@@ -73,4 +63,26 @@ public String calculatePartition(final EventType eventType, final JSONObject eve
         }
     }
 
+    @Override
+    public String calculatePartition(final EventType eventType, final JSONObject event, final List<String> partitions)
+            throws InvalidPartitionKeyFieldsException {
+
+        final List<String> partitionKeyFields = eventType.getPartitionKeyFields();
+        if (partitionKeyFields.isEmpty()) {
+            throw new RuntimeException("Applying " + this.getClass().getSimpleName() + " although event type " +
+                    "has no partition key fields configured.");
+        }
+
+        final int hashValue = extractEventKeys(eventType, event).stream()
+                .map(s -> s.hashCode())
```

Comment:
This is a bug. `s.hashCode()` should not be used, stringHash probably has some code inside, that is not java version dependent.

Reply:
Thanks, I have totally overlooked that.  Don't you think our tests should have helped here?